frontend/src/component/Account/UserSpacesConfig.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { translate } from 'react-i18next'
+import { BtnSwitch, IconButton, ConfirmPopup, ROLE_LIST } from 'tracim_frontend_lib'
+import { newFlashMessage } from '../../action-creator.sync.js'
+import { deleteWorkspaceMember } from '../../action-creator.async.js'
+
+export class UserSpacesConfig extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = { spaceBeingDeleted: null }
+  }
+
+  handleConfirmDeleteSpace = async () => {
+    const { props } = this
+    const spaceId = this.state.spaceBeingDeleted
+    if (!spaceId) return
+    this.setState({ spaceBeingDeleted: null })
+
+    const fetchResult = await props.dispatch(deleteWorkspaceMember(spaceId, props.userToEditId))
+    if (fetchResult.status !== 204) {
+      props.dispatch(newFlashMessage(props.t('Error while leaving the space'), 'warning'))
+    }
+  }
+
+  render = () => {
+    const { props } = this
+
+    const entries = props.workspaceList.map(space => {
```

Comment:
this const could be a separated component

Reply:
Good idea, I'll do this

Reply:
Well, actually this is not so much a component that a list of components that could be empty, so this is a bit tricky to do this.

Reply:
My comment isn't on the right line
I was talking about the `<tr>` code that should be in a separated component

Also, on a second look, instead of returning undefined on some cases and then .filter(entries => entries) afterwards, you could do the filter before the map.

And finally, React doesn't like components that are undefined. That's why you filter them. But it doesn't mind components as `null`, it just renders nothing

Reply:
I need to know whether there are no spaces to show so we can show the "no spaces here" message. I'll try filtering before but I'm not sure this is easy (I would prefer to filter before)

Reply:
I used reduce to avoid the filter and extracted the line in a separate component as requested.